pages/public_cloud/containers_orchestration/managed_kubernetes/migrate-loadbalancer-iolb-to-octavia/guide.en-gb.md

```diff
@@ -0,0 +1,107 @@
+---
+title: How to migrate from LoadBalancer for MKS (IOLB) to Public Cloud LoadBalancer (Octavia)
+excerpt: "How do I migrate from a LoadBalancer for MKS (IOLB) to a LoadBalancer for the Public Cloud (Octavia) and how do I change my DNS"
+updated: 2025-01-17
+---
+
+## Objective
+
+The purpose of this guide is to help OVHcloud Managed Kubernetes Service (MKS) users migrate from their Ingress LoadBalancer (IOLB), which will be deprecated from version MKS 1.32, to the Public Cloud LoadBalancer (Octavia).
+
+It explains the steps required to make this transition safely, minimising service interruptions. Finally, it provides recommendations on DNS management, in particular reducing the TTL, to optimise the propagation of changes and ensure a smooth migration.
+
+> [!warning]
+>
+> As Loadbalancer for Kubernetes and Public Cloud LoadBalancer do not use the same solution for Public IP allocation, it is not possible to keep the existing public IP of your Loadbalancer for Kubernetes. Changing the LoadBalancer class of your Service will lead to the creation of a new Loadbalancer and the allocation of a new Public IP (Floating IP).
+
+## Migration of your LoadBalancer
+
+> [!warning]
+>  
+> Starting with MKS version **1.32**, any upgrade attempt will be blocked if an **IOLB** LoadBalancer service is still present in the cluster.  
+>  
+> Action required**: Before upgrading, you must **migrate your services to the Public Cloud LoadBalancer (Octavia)** by following the steps described in this guide.  
+>  
+> If you attempt to upgrade without first migrating, an error will be returned, preventing the upgrade.
+
+In order to migrate from an existing [LoadBalancer for Kubernetes](/links/public-cloud/load-balancer-kubernetes) to a [Public Cloud LoadBalancer](/links/public-cloud/load-balancer) you will have to modify an existing Service and change its LoadBalancer class.
+
+Your existing LoadBalancer Service using [LoadBalancer for Kubernetes](/links/public-cloud/load-balancer-kubernetes) should have the following annotation:
+
+```yaml
+annotations:
+  loadbalancer.ovhcloud.com/class: "iolb"
+```
+
+##### Step 1 - Edit your Service to change the LoadBalancer class to 'octavia'
+
+```yaml
+annotations:
+  loadbalancer.ovhcloud.com/class: "octavia" // not required for clusters running kubernetes versions >= 1.31, you can just remove the annotation.
+```
+
+##### Step 2 - Apply the change
```

Comment:
at this step the old LB and old IP will be deleted. Customer services are no longer reachable using the old IP, need to perform the DNS switch (provide the new IP immediately).

Note that it is possible to reserve/get the new public IP (floating IP) by advance and then specify it at service/lb creation.